core/includes/common.inc

```diff
@@ -7132,81 +7149,39 @@ function backdrop_sort(array &$array, array $keys = array('weight'), $dir = SORT
   }
   $keys = $new_keys;
 
-  // If sorting on a single value, optimize the sorting to sort only on that
-  // specific key without the overhead of recursive handling.
-  if (count($keys) === 1) {
-    $key = key($keys);
-    $key_sort = reset($keys);
-
-    if ($key_sort === SORT_STRING) {
-      uasort($array, function($a, $b) use ($key, $dir) {
-        if (!is_array($a) || !is_array($b)) {
-          return 0;
-        }
-        if (!isset($a[$key])) {
-          $a[$key] = '';
-        }
-        if (!isset($b[$key])) {
-          $b[$key] = '';
-        }
-        $cmp = strnatcasecmp($a[$key], $b[$key]);
-        return ($dir == SORT_DESC) ? -$cmp : $cmp;
-      });
-    }
-    else {
-      uasort($array, function($a, $b) use ($key, $dir) {
-        if (!is_array($a) || !is_array($b)) {
-          return 0;
-        }
-        if (!isset($a[$key])) {
-          $a[$key] = 0;
-        }
-        if (!isset($b[$key])) {
-          $b[$key] = 0;
-        }
-        if ($a[$key] != $b[$key]) {
-          $cmp = $a[$key] < $b[$key] ? -1 : 1;
-          return ($dir == SORT_DESC) ? -$cmp : $cmp;
-        }
-        // Equal values, return 0.
-        return 0;
-      });
-    }
-  }
-  // If doing a multiple-key comparison, use a recursive callback.
-  else {
-    $keys_map = array_keys($keys);
-    $recursive_callback = function ($a, $b, $key_index = 0) use ($keys, $keys_map, $dir, &$recursive_callback) {
-      $key = $keys_map[$key_index];
-      if (!is_array($a) || !is_array($b)) {
-        return 0;
-      }
-      if (!isset($a[$key])) {
-        $a[$key] = 0;
-      }
-      if (!isset($b[$key])) {
-        $b[$key] = 0;
-      }
-      if ($a[$key] == $b[$key]) {
-        // Sort on the next key if one exists.
-        if (isset($keys_map[$key_index + 1])) {
-          return $recursive_callback($a, $b, $key_index + 1);
-        }
+  // Create an array to use for sorting by each key.
+  $sort = array();
```

Comment:
Above `$sort` is assigned a constant (ie. an integer) but here it becomes an array. I'm guessing one or the other variable needs to be renamed.

Reply:
Depends... Are you worried about the variable types getting mixed up, or just the readability of the code?
If the former, no worries as the first `$sort` is scoped to the `foreach()` statement, while the second is outside it.
If the latter, I guess it might be confusing for some, but the first `$sort` has done its job and is then re-assigned for a different purpose further on in the function...

Reply:
Alright, I get it. I didn't see that you assigned it to `$new_key` so you don't need it anymore. Maybe for legibility they could be different.

Reply:
New commit changes previous `$sort` to be consistent with the rest of the code (i.e. all `foreach()` iterating over the `$keys` array now uses `$key` and `$key_sort`. All iterations of the `$array` array use `$key` and `$value` (with the exception of one where that'd conflict with the `$keys` array iteration so uses `$k` and `$v` instead)).